### PR TITLE
[RPS-103] Add pagination to search results

### PR DESCRIPTION
**Pull request by ngta1**
Includes standard next and prev links, in addition to
first and last links. Set to 10 results per page, with
the number configurable in console.

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/css/demo.css [hunk 1]
```diff
@@ -96,3 +96,136 @@ input {
...
+
+/* Pagination styling from Bootstrap css. To replace when styling is defined for site */
+.pagination {
+    display: inline-block;
+    padding-left: 0;
+    margin: 20px 0;
+    border-radius: 4px;
+}
+
+.pagination > li {
```
Comment (woledzki) [inline repository-data/webfiles/src/main/resources/site/css/demo.css]: no need for ">" here and anywhere in this CSS

File: repository-data/webfiles/src/main/resources/site/css/demo.css [hunk 2]
```diff
@@ -96,3 +96,136 @@ input {
...
+.pagination > .disabled > a,
+.pagination > .disabled > a:hover,
+.pagination > .disabled > a:focus {
+    color: #999999;
+    cursor: not-allowed;
+    background-color: #ffffff;
+    border-color: #dddddd;
+}
+
+.pagination-lg > li > a,
```
Comment (woledzki) [inline repository-data/webfiles/src/main/resources/site/css/demo.css]: large and small variants not used anywhere. Delete please.

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java [hunk 3]
```diff
@@ -13,48 +11,59 @@
...
 import org.hippoecm.hst.util.SearchInputParsingUtils;
+import org.onehippo.cms7.essentials.components.EssentialsSearchComponent;
+import org.onehippo.cms7.essentials.components.info.EssentialsListComponentInfo;
+import org.onehippo.cms7.essentials.components.paging.Pageable;
 import uk.nhs.digital.ps.beans.Publication;
 
 import static org.hippoecm.hst.content.beans.query.builder.ConstraintBuilder.constraint;
 import static org.hippoecm.hst.content.beans.query.builder.ConstraintBuilder.or;
 
-@ParametersInfo(type = SearchComponentInfo.class)
```
Comment (ziemsky) [inline site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: `SearchComponentInfo` appears to no longer be used - can it be deleted?

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java [hunk 4]
```diff
@@ -13,48 +11,59 @@
...
         HstRequestContext requestContext = request.getRequestContext();
         HippoBean scopeBean = requestContext.getContentBean();
-        SearchComponentInfo info = getComponentParametersInfo(request);
+        final EssentialsListComponentInfo paramInfo = getComponentParametersInfo(request);
         String query = SearchInputParsingUtils.parse(
             getPublicRequestParameter(request, "query"), false);
-        int offset = NumberUtils.toInt(
-            getPublicRequestParameter(request, "offset"), 0);
+        final int pageSize = getPageSize(request, paramInfo);
+        final int currentPage = getCurrentPage(request);
```
Comment (ziemsky) [inline site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: `currentPage` suggests that this variable represents some object of type `Page`. How about renaming it to something like `currentPageNumber`, `currentPageIndex` or `currentPageOrdinal`?

File: repository-data/webfiles/src/main/resources/site/css/demo.css (hunk 1 above)
Comment (ngta1) [reply repository-data/webfiles/src/main/resources/site/css/demo.css]: Will remove from css

File: repository-data/webfiles/src/main/resources/site/css/demo.css (hunk 2 above)
Comment (ngta1) [reply repository-data/webfiles/src/main/resources/site/css/demo.css]: good spot. removed also.

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java (hunk 3 above)
Comment (ngta1) [reply site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: Yes, no longer needed so will remove

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java (hunk 4 above)
Comment (ngta1) [reply site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: Good idea. will go for currentPageNumber

**Comment by ngta1** [comment]
Thanks both.  Changes applied.

**Pull request merged**